tests/misc/iterable_sections/src/main.c

```diff
@@ -98,6 +119,12 @@ const STRUCT_SECTION_ITERABLE_NAMED(test_rom_named, C, rom7) = {0x30};
 const STRUCT_SECTION_ITERABLE_NAMED(test_rom_named, D, rom8) = {0x40};
 const STRUCT_SECTION_ITERABLE_NAMED(test_rom_named, B, rom9) = {0x20};
 
+/* declare in random order to check that the linker is sorting numerically */
+const STRUCT_SECTION_ITERABLE(test_rom_numeric, rom10) = {0x10};
+const STRUCT_SECTION_ITERABLE(test_rom_numeric, rom12) = {0x30};
```

Comment:
Can  you add a number that is 0-9 in this test to make sure its sorted correctly.